wadas/ui/mainwindow.py

```diff
@@ -19,19 +19,25 @@
 
 
 import datetime
+import io
 import logging
 import os
 import sys
 from datetime import timedelta
 from logging.handlers import RotatingFileHandler
+
+from collections import deque
```

Comment:
collections is part of the stdlib: remove the blank line, and move this line at L26.